machineconfiguration/v1alpha1/types_osimagestream.go

```diff
@@ -0,0 +1,126 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=osimagestreams,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2555
+// +openshift:file-pattern=cvoRunLevel=0000_80,operatorName=machine-config,operatorOrdering=01
+// +openshift:enable:FeatureGate=OSStreams
+// +kubebuilder:metadata:labels=openshift.io/operator-managed=
+
+// OSImageStream describes a set of streams and associated URLs available
+// for the MachineConfigPools to be used as base OS images.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type OSImageStream struct {
```

Comment:
We can maybe add a validation like https://github.com/openshift/api/blob/master/operator/v1/types_olm.go#L22 to enforce a singleton for the cluster?